### PR TITLE
gpu: Improve installation options and documentation

### PR DESCRIPTION
**Pull request by tkatila**
Signed-off-by: Tuomas Katila <tuomas.katila@intel.com>

### CODE REVIEW

File: cmd/gpu_plugin/README.md [hunk 1]
```diff
@@ -57,27 +57,34 @@ Release tagged images of the components are also available on the Docker hub, ta
...
 
-Where `<RELEASE_VERSION>` needs to be substituted with the desired [release tag](https://github.com/intel/intel-device-plugins-for-kubernetes/tags) or `main` to get `devel` images.
+#### Install to nodes with Intel GPUs with NFD
 
-Alternatively, if your cluster runs
-[Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery),
-you can deploy the device plugin only on nodes with Intel GPU.
-The [nfd_labeled_nodes](../../deployments/gpu_plugin/overlays/nfd_labeled_nodes/)
-kustomization adds the nodeSelector to the DaemonSet:
+Deploying GPU plugin to only nodes that have Intel GPU attached. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) is required to detect the presense of Intel GPUs.
```
Comment (uniemimu) [inline cmd/gpu_plugin/README.md:74]:
```suggestion
Deploying GPU plugin to only nodes that have Intel GPU attached. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) is required to detect the presence of Intel GPUs.
```

**Review by uniemimu — APPROVED** [review]
lgtm

**Review by hj-johannes-lee — APPROVED** [review]
LGTM!

**Review by mythi — APPROVED** [review]

**Comment by eero-t** [comment]
@tkatila This changed / added new sections but did not update TOC.  As this got merged, please add new PR updating TOC too.

**Comment by tkatila** [comment]
> @tkatila This changed / added new sections but did not update TOC. As this got merged, please add new PR updating TOC too.

Thanks for pointing this out. I didn't realize that.

**Pull request merged**